src/routes/documents/projects/[id]-[slug]/+page.ts

```diff
@@ -1,20 +1,47 @@
-import * as projectApi from "$lib/api/projects";
+import { error, redirect } from "@sveltejs/kit";
+
+import { DEFAULT_PER_PAGE } from "@/config/config.js";
+import * as projects from "$lib/api/projects";
+import * as collaborators from "$lib/api/collaborators";
 import { search } from "$lib/api/documents";
 import { breadcrumbTrail } from "$lib/utils/navigation";
 
 export async function load({ params, url, parent, fetch }) {
   const id = parseInt(params.id);
-  const project = await projectApi.get(id, fetch);
+  const [project, users] = await Promise.all([
+    projects.get(id, fetch).catch(console.error),
+    collaborators.list(id, fetch).catch(console.error),
+  ]).catch((e) => {
+    return [];
+  });
+
+  if (!project) {
+    error(404, "Project not found");
+  }
+
+  if (project.slug !== params.slug) {
+    const canonical = projects.canonicalUrl(project);
+    redirect(302, canonical);
+  }
```

Comment:
We should probably handle this redirection first-thing, before waiting for project and collaborator requests to resolve.

Reply:
I don't think we can do this without doing the requests in sequence. I'm going to leave this as-is right now and plan on doing an optimization pass on all of our data loading logic later this week.